src/main/java/expression/tokenextractor/AbstractFixedValueTokenExtractor.java

```diff
@@ -0,0 +1,20 @@
+package expression.tokenextractor;
+
+import expression.token.FixedValueToken;
+import expression.token.Token;
+
+public abstract class AbstractFixedValueTokenExtractor implements TokenExtractor {
+
+  protected abstract FixedValueToken getToken();
+
+  @Override
+  public boolean startsWithSupportedToken(String expression, int startNdx) {
```

Comment:
It might be good to enforce the call to this method (by creating another method that calls `startsWithSupportedToken(...)` and `extractToken(...)` in sequence) instead of relying on the client of this class to call it before calling `extractToken(...)`